### PR TITLE
fix: Clarify what has-root-span does

### PR DESCRIPTION
**Pull request by cartermp**
Noted in pollinators: https://honeycombpollinators.slack.com/archives/C01BZ3J7RMJ/p1714601065818559

### CODE REVIEW

File: rules_conditions.md [hunk 1]
```diff
@@ -201,6 +201,15 @@ For most cases, use `not-exists` in a rule with a scope of "span".
 WARNING: Rules can have `Scope: trace` or `Scope: span`; `not-exists` used with `Scope: trace` will be true if **any** single span in the entire trace matches the negative condition.
 This is almost never desired behavior.
 
+### has-root-span
+
+Tests if the trace as a whole has a root span.
+
+The `Value` parameter can either be `true` or `false`.
+
+**NOTE:** `has-root-span` does not check if a given span is a root span.
```
Comment (VinozzZ) [inline rules_conditions.md:210]: nit - it looks like other places uses `WARNING` for calling out special characteristic of rule conditions. Maybe we can change `**NOTE**` to `WARNING:` here as well

**Comment by cartermp** [comment]
@VinozzZ done!

File: rules_conditions.md [hunk 2]
```diff
@@ -201,6 +201,15 @@ For most cases, use `not-exists` in a rule with a scope of "span".
...
 This is almost never desired behavior.
 
+### has-root-span
+
+Tests if the trace as a whole has a root span.
+
+The `Value` parameter can either be `true` or `false`.
+
+WARNING: `has-root-span` does not check if a given span is a root span.
+It only checks if a trace has a root span.
```
Comment (kentquirk) [inline rules_conditions.md:211]:
```suggestion
NOTE: `has-root-span` does not check if a given span **is** a root span,
it checks if the containing trace **has** a root span.
```

File: rules_complete.yaml [hunk 3]
```diff
@@ -108,6 +108,8 @@ Samplers:
                 - Name: drop incomplete traces from the buggy service
                   Drop: true
                   Conditions:
+                    # NOTE: has-root-span only checks if the trace has a root span.
+                    # It does NOT act as a short-hand for evaluating the root span.
```
Comment (kentquirk) [inline rules_complete.yaml:112]:
```suggestion
                    # It does NOT act as a shorthand for evaluating the root span; use the `root.` construct for that.
```

**Review by kentquirk — COMMENTED** [review]
I tweaked it a bit. What do you think?

File: rules_conditions.md (hunk 2 above)
Comment (kentquirk) [reply rules_conditions.md:211]: Huh, my suggestion includes the opposite of Yingrong's -- I feel like "WARNING" is too strident and want this to be a note.

**Review by kentquirk — APPROVED** [review]

**Comment by danielcompton** [comment]
Thank you! 😊

**Pull request merged**